providers-sdk/v1/upstream/mvd/cvss/cvss_test.go

```diff
@@ -141,6 +141,57 @@ func TestCvss3ParseEmpty(t *testing.T) {
 	assert.Nil(t, c, "no object returned")
 }
 
+func TestCvss4Parsing(t *testing.T) {
+	c, err := New("9.3/CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:H/VI:H/VA:H/SC:H/SI:H/SA:H")
+	assert.Nil(t, err, "could parse the cvss vector")
+	assert.True(t, c.Verify(), "valid cvss vector")
+
+	assert.Equal(t, float32(9.3), c.Score, "score properly detected")
+	assert.Equal(t, "Critical", c.Severity().String(), "severity properly extracted")
+	assert.Equal(t, "4.0", c.Version(), "vector format version")
+
+	metrics, err := c.Metrics()
+	assert.Nil(t, err, "metrics could be extracted")
+
+	// Base Metrics
+	assert.Equal(t, "N", metrics["AV"], "AV properly detected")
+	assert.Equal(t, "L", metrics["AC"], "AC properly detected")
+	assert.Equal(t, "N", metrics["AT"], "AT properly detected")
+	assert.Equal(t, "N", metrics["PR"], "PR properly detected")
+	assert.Equal(t, "N", metrics["UI"], "UI properly detected")
+
+	// Vulnerable System Impacts
+	assert.Equal(t, "H", metrics["VC"], "VC properly detected")
+	assert.Equal(t, "H", metrics["VI"], "VI properly detected")
+	assert.Equal(t, "H", metrics["VA"], "VA properly detected")
+
+	// Subsequent System Impacts
+	assert.Equal(t, "H", metrics["SC"], "SC properly detected")
+	assert.Equal(t, "H", metrics["SI"], "SI properly detected")
+	assert.Equal(t, "H", metrics["SA"], "SA properly detected")
+}
+
+func TestCvss4Parsing2(t *testing.T) {
```

Comment:
Please also add a test where the CVSS vector includes all possible metrics.

Reply:
Added